php/libraries/Utility.class.inc

```diff
@@ -1277,5 +1277,32 @@ class Utility
         }
         return checkdate($dateElement['M'], $dateElement['d'], $dateElement['Y']);
     }
+
+    /**
+     * Determine if a string starts with another string
+     *
+     * @param string $haystack string to search
+     * @param string $needle   substring to find
+     *
+     * @return boolean if $haystack begins with $needle
+     */
+    static function startsWith($haystack, $needle)
+    {
+        return substr($haystack, 0, strlen($needle)) === $needle;
+    }
+
+    /**
+     * Determine if a string ends with another string
+     *
+     * @param string $haystack string to search
+     * @param string $needle   substring to find
+     *
+     * @return boolean if $haystack ends with $needle
+     */
+    static function endsWith($haystack, $needle)
+    {
+        return substr($haystack, -strlen($needle)) === $needle;
+    }
```

Comment:
I think these would make more sense in a new Strings class, rather than adding more stuff to the already overloaded Utility class.